### PR TITLE
KTOR-6596 Fix misleading readBytes() function name

### PR DESCRIPTION
**Pull request by e5l**
Fix [KTOR-6596](https://youtrack.jetbrains.com/issue/KTOR-6596) Misleading function name

### CODE REVIEW

File: ktor-client/ktor-client-core/common/src/io/ktor/client/statement/HttpResponse.kt [hunk 1]
```diff
@@ -50,17 +48,36 @@ public abstract class HttpResponse : HttpMessage, CoroutineScope {
     public abstract val responseTime: GMTDate
 
     /**
-     * Unmodified [ByteReadChannel] with the raw payload of the response.
+     * Provides a raw [ByteReadChannel] to the response content as it was read from the network.
```
Comment (vnikolova) [inline ktor-client/ktor-client-core/common/src/io/ktor/client/statement/HttpResponse.kt:51]:
```suggestion
     * Provides a raw [ByteReadChannel] to the response content as it is read from the network.
```

File: ktor-client/ktor-client-core/common/src/io/ktor/client/statement/HttpResponse.kt [hunk 2]
```diff
@@ -92,6 +108,14 @@ public suspend fun HttpResponse.bodyAsText(fallbackCharset: Charset = Charsets.U
 }
 
 /**
- * Reads the [HttpResponse.content] as a [ByteReadChannel].
+ * Reads the [HttpResponse.rawContent] as a [ByteReadChannel].
  */
 public suspend fun HttpResponse.bodyAsChannel(): ByteReadChannel = body()
+
+/**
+ * Reads the response body as a byte array. All the plugins will be applied to the response body, so it may be
```
Comment (vnikolova) [inline ktor-client/ktor-client-core/common/src/io/ktor/client/statement/HttpResponse.kt:116]:
```suggestion
 * Reads the response body as a byte array. Note that all plugins will be applied to the response body, which may be
```

File: ktor-client/ktor-client-core/common/src/io/ktor/client/statement/Readers.kt [hunk 3]
```diff
@@ -5,29 +5,49 @@
...
 /**
- * Reads the whole [HttpResponse.content] if `Content-Length` is specified.
- * Otherwise, it just reads one byte.
+ * Reads the raw payload of the HTTP response as a byte array.
+ *
+ * This method reads the raw payload of the HTTP response as a byte array.
+ * The raw payload is the content
+ * of the response that hasn't gone through any interceptors from the HttpResponsePipeline.
+ * The content will keep
+ * compression or encoding as it was received from the server.
```
Comment (vnikolova) [inline ktor-client/ktor-client-core/common/src/io/ktor/client/statement/Readers.kt:25]:
```suggestion
 * compression or encoding as received from the server.
```

File: ktor-client/ktor-client-core/common/src/io/ktor/client/statement/Readers.kt [hunk 4]
```diff
@@ -5,29 +5,49 @@
...
 
 /**
- * Reads the whole [HttpResponse.content] if `Content-Length` is specified.
- * Otherwise, it just reads one byte.
+ * Reads the raw payload of the HTTP response as a byte array.
+ *
+ * This method reads the raw payload of the HTTP response as a byte array.
+ * The raw payload is the content
+ * of the response that hasn't gone through any interceptors from the HttpResponsePipeline.
+ * The content will keep
```
Comment (vnikolova) [inline ktor-client/ktor-client-core/common/src/io/ktor/client/statement/Readers.kt:24]:
```suggestion
 * The content will retain its original
```

File: ktor-client/ktor-client-core/common/src/io/ktor/client/statement/Readers.kt [hunk 5]
```diff
@@ -5,29 +5,49 @@
...
  */
+@OptIn(InternalAPI::class)
+public suspend fun HttpResponse.readRawBytes(): ByteArray = rawContent.readRemaining().readByteArray()
 
+/**
+ * Reads the raw payload of the HTTP response as a byte array.
+ *
+ * This method reads the raw payload of the HTTP response as a byte array.
+ * The raw payload is the content of the response that hasn't gone through any interceptors.
+ * The content will keep compression or encoding as it was received from the server.
```
Comment (vnikolova) [inline ktor-client/ktor-client-core/common/src/io/ktor/client/statement/Readers.kt:37]:
```suggestion
 * The content will retain its original compression or encoding as received from the server.
```

**Review by vnikolova — APPROVED** [review]
Just some small edits, otherwise looks clear!

**Comment by e5l** [comment]
@vnikolova, thank you for the review!

**Pull request merged**